src/app/pages/components/editor/editor-context.tsx

```diff
@@ -9,34 +9,41 @@ import {
     DropdownItem, Gallery, GalleryItem,
     MenuToggle,
     Text,
-    TextContent
-} from "@patternfly/react-core";
-import {If, NavLink, ToggleIcon} from "@app/components";
+    TextContent} from "@patternfly/react-core";
+import { If, NavLink, ToggleIcon } from "@app/components";
 import Moment from "react-moment";
-import {LocalStorageService, useLocalStorageService} from "@app/services";
 import {DesignContext} from "@app/models/designs/design-context.model";
 import {ExportToRhosrData, ExportToRhosrModal} from "@app/pages/components";
+import { LocalStorageService, useLocalStorageService } from "@app/services";
+import { Registry } from "@rhoas/registry-management-sdk";
+import { RegistryDryRunFormModal } from "./dry-run.modal";
+import { DryRunErrorResponse } from "@app/pages/editor";
 
 /**
  * Properties
  */
 export type EditorContextProps = {
     design: Design;
     dirty: boolean;
+    artifactContent: string;
     onSave: () => void;
     onCancel: () => void;
+    onExpandDryRunCausesPanel: (error: DryRunErrorResponse) => void;
+    onRegistrationDryRun: (registry: Registry, group: string | undefined, artifactId: string) => void;
+    isPanelOpen?: boolean;
 }
 
 /**
  * The context of the design when editing a design on the editor page.
  */
-export const EditorContext: FunctionComponent<EditorContextProps> = ({design, dirty, onSave, onCancel}: EditorContextProps) => {
+export const EditorContext: FunctionComponent<EditorContextProps> = ({ design, dirty, onSave, onRegistrationDryRun, onExpandDryRunCausesPanel }: EditorContextProps) => {
     const lss: LocalStorageService = useLocalStorageService();
 
     const [designContext, setDesignContext] = useState<DesignContext>();
     const [isActionMenuToggled, setActionMenuToggled] = useState(false);
     const [isExpanded, setExpanded] = useState(lss.getConfigProperty("editor-context.isExpanded", "false") === "true");
     const [isRegisterModalOpen, setRegisterModalOpen] = useState(false);
+    const [isDryRunModalOpen, setIsDryRunModalOpen] = useState(false);
```

Comment:
Not a big deal but generally I like to not have the "Is" in the setter.  So `isDirty` and `setDirty` rather than `isDirty` and `isSetDirty`.  Not asking you to change it, just offering it as a suggestion if you like it.  This is actually a common pattern in Java, where in a Java Bean (simple java model class) if you have a field named e.g. `dirty` with type `boolean`, the convention is to name your getter `isDirty()` and your setter `setDirty()`.

FWIW